python/hsml/core/explicit_provenance.py

```diff
@@ -0,0 +1,338 @@
+        exception_cause=None,
+        **kwargs,
+    ):
+        self._model_registry_id = model_registry_id
+        self._name = name
+        self._version = version
+        self._type = type
+        self._meta_type = meta_type
+        self._href = href
+        self._exception_cause = exception_cause
+
+    @property
+    def model_registry_id(self):
+        """Id of the model registry in which the artifact is located."""
+        return self._model_registry_id
+
+    @property
+    def name(self):
+        """Name of the artifact."""
+        return self._name
+
+    @property
+    def version(self):
+        """Version of the artifact"""
+        return self._version
+
+    def __str__(self):
+        return {
+            "model_registry_id": self._model_registry_id,
+            "name": self._name,
+            "version": self._version,
+        }
+
+    def __repr__(self):
+        return (
+            f"Artifact({self._model_registry_id!r}, {self._name!r}, "
+            f"{self._version!r}, {self._type!r}, {self._meta_type!r}, "
+            f"{self._href!r}, {self._exception_cause!r})"
+        )
+
+    @staticmethod
+    def from_response_json(json_dict: dict):
+        link_json = humps.decamelize(json_dict)
+        if link_json.get("exception_cause") is not None:
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.FAULTY,
+            )
+        elif bool(link_json["deleted"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.DELETED,
+            )
+        elif not bool(link_json["accessible"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.INACCESSIBLE,
+                link_json["artifact"]["href"],
+            )
+        elif bool(link_json["accessible"]):
+            return Artifact(
+                link_json["artifact"]["project"],
+                link_json["artifact"]["name"],
+                link_json["artifact"]["version"],
+                link_json["artifact_type"],
+                Artifact.MetaType.NOT_SUPPORTED,
+                link_json["artifact"]["href"],
+            )
+        else:
+            return None
+
+
+class Links:
+    def __init__(self):
+        self._accessible = []
+        self._deleted = []
+        self._inaccessible = []
+        self._faulty = []
+
+    @property
+    def deleted(self):
+        """List of [Artifact objects](../links#artifact) which contains
```

Comment:
Does this ../links refer to something valid?

Reply:
updated and removed.